library/src/main/java/com/getstream/sdk/chat/StreamChat.java

```diff
@@ -229,9 +249,24 @@ public void onError(String errMsg, int errCode) {
                         onlineStatus.postValue(OnlineStatus.FAILED);
                     }
                 });
+                registerFirebaseToken(context);
             }
             return true;
         }
     }
 
+    private static void registerFirebaseToken(Context context) {
```

Comment:
Looks like the whole method should be moved to `NotificationsManager`.  `StreamChat` doesn't really care about Firebase details, so we might move all notifications responsibilities into one place.

```kotlin
class NotificationsManager (val context: Context) {
  init(){
     FirebaseInstanceId.getInstance().getInstanceId()
                .addOnCompleteListener(task -> {
                            if (task.isSuccessful() && task.getResult() != null)
                                setFirebaseToken(task.getResult().getToken(), context);
                        }
    ); 
  }
}

notificationsManager.init()
```
`StreamChat` also becomes complicated. It should work with modules like `NotificationsManager` or `FontsManager`, and contain less implementation details.